Find a parent when CWD disappears

0xACE opened the pull request:
The previous behaviour would exit nnn when the CWD disappeared.

0xACE commented:
The failed tests seems to be because of syntax complaints. It's very unclear which syntax style you use, but I found `} else if (...) {` used elsewhere so I imitated it. Personally I wouldn't use that either and just imitated the existing style.

Otherwise I have tested the code. The code also makes the assumption that `/` is always going to be an accessible path. If there are situations where `/` isn't a accessible path then I don't know what the program should do, but you could check for it with `isatopdir() && access() == -1` to detect this situation.

jarun commented on src/nnn.c:
```diff
@@ -3297,11 +3297,36 @@ static void browse(char *ipath)
+			xstrlcpy(path, dir, PATH_MAX);
+
+			setdirwatch();
+			mvprintw(xlines - 1, 0, "folder disappeared\n");
```
Please use the message: "cannot access directory"

jarun commented on src/nnn.c:
```diff
@@ -3297,11 +3297,36 @@ static void browse(char *ipath)
+					if (!dir)
+						dir = dirname(newpath);
+					break;
+				} else if (!dir) {
```
Please remove `else if` and make it a separate `if` clause. (You don't need `else if` after a `break`).

jarun replied:
This is also the reason CI fails.

jarun commented on src/nnn.c:
```diff
@@ -3297,24 +3297,19 @@ static void browse(char *ipath)
-			xstrlcpy(lastdir, path, PATH_MAX);
-
-			/* Save history */
-			xstrlcpy(lastname, xbasename(path), NAME_MAX + 1);
```
Please keep the lastname. The directory can stay selected in parent even if the user doesn't have permissions anymore.

jarun replied:
If the dir is deleted, we select the top entry anyway.

0xACE commented:
You were quick to respond :).

All your comments has been addressed.

jarun commented:
> You were quick to respond :).

Thanks to a reasonably powerful Android phone and Termux! ;)

Thank you for your contribution!

Please take a look at the scroll support with ncurses6. It would be a great value addition for our users on Termux. I am one of them! ;)

jarun commented:
Frankly speaking, I am kind of obsessed by Termux! (That's a compliment for you @fornwall ;))

@0xACE if you have any queries regarding the Termux environment @fornwall is the goto guy.

0xACE commented:
> Please take a look at the scroll support with ncurses6. It would be a great value addition for our users on Termux. I am one of them! ;)

I am also a termux user, i addressed your comments from the phone and didn't push the changes until i reached a computer, termux is super nice when I'm on the run.

Regarding termux scroll support I would suggest a specific github issue so that the problem can be tracked from 1 place. As, i would also argue that in termux a "click" should be on the "highlighted" item rather then where the user clicks as it's hard to click on the exact row...

Yeah, frankly from termux I really wish they had this fixed sooner: https://github.com/termux/termux-api/pull/233

My early guess is that it's just a matter of implementing the code, and not a very termux specific problem...

jarun commented:
> Regarding termux scroll support I would suggest a specific github issue

We will continue the discussion in the ToDo list for now.
